interpreter/prune_test.go

```diff
@@ -466,9 +468,22 @@ func TestPrune(t *testing.T) {
 			ExhaustiveEval(), Observe(EvalStateObserver(state)))
 		interpretable.Eval(testActivation(t, tst.in))
 		newExpr := PruneAst(ast.GetExpr(), ast.GetSourceInfo().GetMacroCalls(), state)
+		if tst.iterRange != "" {
```

Comment:
Up to you: since this check only applies for a single test, consider pulling this out into a separate test.

Reply:
I added some additional tests. PTAL